### PR TITLE
feat(material-experimental): add autocomplete test harness

### PR DESCRIPTION
**Pull request by crisbeto**
* Adds a test harness for `mat-autocomplete`.
* Fixes the global element locator not working for `TestBed` elements.
* Adds the ability to get a property of a `TestElement`.

### CODE REVIEW

File: src/material-experimental/mdc-autocomplete/harness/autocomplete-harness.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    return this._options();
+  }
+
+  /** Gets the groups of options inside the panel. */
+  async getOptionGroups(): Promise<TestElement[]> {
+    return this._groups();
+  }
+
+  /** Gets whether the autocomplete panel is visible. */
+  async panelIsVisible(): Promise<boolean> {
```
Comment (jelbourn) [inline src/material-experimental/mdc-autocomplete/harness/autocomplete-harness.ts]: `isPanelVisible`?

**Comment by jelbourn** [comment]
@crisbeto was this one still blocked on anything?

**Comment by crisbeto** [comment]
I don't think it is @jelbourn, I just need to rebase it.

**Comment by crisbeto** [comment]
Rebased @jelbourn.

**Comment by crisbeto** [comment]
The discussion around `getAttribute` and `getProperty` was resolved by https://github.com/angular/components/pull/16706. I've addressed the rest of the comments as well.

**Pull request merged**